lockbox-ios/Action/DataStoreAction.swift

```diff
@@ -362,13 +366,15 @@ extension DataStoreActionHandler {
                     .entry(ItemEntry.Builder()
                             .kind("login")
                             .username("tjacobson@yahoo.com")
+                            .password("iLUVdawgz")
```

Comment:
err.. @linuxwolf ?

Reply:
_"can"_ 🤔  or _"should (not)"_ 🤦‍♀️ ?

Remember that these values are ultimately what a user got the website they care about to accept.  There are character choices that are not a great idea, but putting any other restrictions can be limiting user choice and possibly discriminatory:

* **spaces** -- [diceware recommends spaces](http://world.std.com/%7Ereinhold/diceware.html)
* **emoji** -- bad for [system logins](https://apple.stackexchange.com/questions/202143/i-included-emoji-in-my-password-and-now-i-cant-log-in-to-my-account-on-yosemite), but actually not for web logins (if the site accepts them) 🤷‍♂️ 
* **RTL** strings -- would bias against Hebrew, Arabic, and some other Asiatic natives
* empty strings -- well, we probably should  enforce something, but this might have to be reserved for the data-entry front-ends.

Reply:
Also, this is ultimately handled by the [datastore](/mozilla-lockbox/lockbox-datastore);  performing any further restrictions over what it does effectively means user data loss.

Reply:
I'm more interested in the behaviour of the UI. 

Can the UI handle with whatever passwords the datastore can store? 

If the datastore is handling weird passwords, then the UI should too. Tests should reflect that.

Sorry for the confusion.